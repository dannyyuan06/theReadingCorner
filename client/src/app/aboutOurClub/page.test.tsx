import { render } from "@testing-library/react";
import '@testing-library/jest-dom'
import AboutOurClub from "./page";

describe("AboutOurClub Component", () => {
  it("renders the component without errors", () => {
    const { getByText } = render(<AboutOurClub />);
    
    // Replace with appropriate test assertions
    expect(getByText("ABOUT OUR CLUB")).toBeInTheDocument();
    expect(getByText("We envision a world where literacy is the catalyst for change. We're here to make a difference.")).toBeInTheDocument();
    // Add more assertions as needed
  });
  
  it("matches snapshot", () => {
    const { asFragment } = render(<AboutOurClub />);
    
    // Use Jest's snapshot testing to compare component output with a saved snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});