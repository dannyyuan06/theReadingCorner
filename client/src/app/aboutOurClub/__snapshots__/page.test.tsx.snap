// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutOurClub Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="header"
    >
      ABOUT OUR CLUB
    </h1>
    <section>
      <p>
        The Reading Corner is a dedicated organisation committed to creating diverse and meaningful literary opportunities and experiences within our community. We strive to bridge the gap between literature and underprivileged communities, ensuring that everyone has access to the transformative power of books and storytelling.
      </p>
      <p>
        We aspire to create a community where literature is seen as a catalyst for personal and societal transformation, where individuals are inspired to explore new ideas, cultivate empathy, and embrace the beauty of diverse perspectives.
      </p>
    </section>
    <section
      class="splitContainer"
    >
      <div
        class="splitWrapper splitWrapperBold"
      >
        <h2>
          We envision a world where literacy is the catalyst for change. We're here to make a difference.
        </h2>
        <a
          class="signUpForNews buttons"
          href="https://www.thereadingcorner.uk/lead-collection"
        >
          SIGN UP FOR NEWS AND OPPORTUNITIES
        </a>
      </div>
      <div
        class="splitWrapper"
      >
        <p
          style="margin-top: 0px;"
        >
          The Reading Corner is a dedicated organisation that strives to bridge the literacy gap among marginalized youth in schools, hospitals, and underserved communities. By providing support, we empower children and young adults to discover stories and authors that resonate with them, ultimately fostering a lifelong love of reading.
        </p>
        <p>
          We believe that access to reading materials and the ability to relate to stories and characters can make a significant impact on a child's educational and personal development. Our mission is to make reading accessible and enjoyable for all, regardless of their circumstances.
        </p>
      </div>
    </section>
    <section>
      <div
        class="splitContainer lastSection"
      >
        <img
          alt="our goals photo"
          data-nimg="1"
          decoding="async"
          height="540"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fphoto_our_goals.webp&w=1080&q=75"
          srcset="/_next/image?url=%2Fimages%2Fphoto_our_goals.webp&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fphoto_our_goals.webp&w=1080&q=75 2x"
          style="color: transparent;"
          width="447"
        />
        <div
          class="splitWrapper"
          style="padding-top: 0px;"
        >
          <h2>
            OUR GOALS
          </h2>
          <p
            style="margin-top: 0px;"
          >
            The Reading Corner is an ambitious non-profit organization founded on the belief that literature has the power to change the world. Our mission is fueled by a commitment to the following long-term goals:
          </p>
          â€‹ 
          <ul
            class="goalList"
          >
            <li>
              We're defining a new generation of literacy access.
            </li>
            <li>
              We foster a shared love of learning and knowledge.
            </li>
            <li>
              We empower people to reach their fullest potential.
            </li>
            <li>
              We create meaningful opportunities in literacy for all.
            </li>
            <li>
              We empower the next generation of social change-makers.
            </li>
          </ul>
          <a
            class="buttons learnMore"
            href=""
          >
            Learn more about our strategy
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
